modules/v2x/v2x_proxy/proto_adapter/proto_adapter.cc

```diff
@@ -0,0 +1,230 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/v2x/v2x_proxy/proto_adapter/proto_adapter.h"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "cyber/cyber.h"
+
+namespace v2x {
+namespace proxy {
+
+OSLightype AdapterLightTypeObu2Sys(int32_t type) {
+  switch (type) {
+    case 2:
+      return OSLightype::SingleTrafficLight_Type_LEFT;
+    case 3:
+      return OSLightype::SingleTrafficLight_Type_RIGHT;
+    case 4:
+      return OSLightype::SingleTrafficLight_Type_U_TURN;
+    case 1:
+    default:
+      return OSLightype::SingleTrafficLight_Type_STRAIGHT;
+  }
+}
+
+bool AdapterLightObu2Sys(const OBULight &obu_light,
+                         std::shared_ptr<OSLight> *os_light_ptr) {
+#if 0
+  if (nullptr == os_light_ptr) {
+    return false;
+  }
+  auto res = std::make_shared<OSLight>();
+  if (nullptr == res) {
+    return false;
+  }
+  if (!obu_light.has_header()) {
+    return false;
+  }
+  if (obu_light.header().has_timestamp_sec()) {
+    res->mutable_header()->set_timestamp_sec(
+        obu_light.header().timestamp_sec());
+  }
+  if (obu_light.header().has_module_name()) {
+    res->mutable_header()->set_module_name(obu_light.header().module_name());
+  }
+  if (obu_light.road_trafficlight_size() < 1) {
+    return false;
+  }
+  //// FOR-EACH ROAD
+  bool flag_has_data = false;
+  for (int idx_road = 0; idx_road < obu_light.road_trafficlight_size();
+       idx_road++) {
+    const auto &obu_road_light1 = obu_light.road_trafficlight(idx_road);
+    // Set the road index for lane
+    apollo::v2x::RoadTrafficLight_Attribute tl_attr =
+        apollo::v2x::RoadTrafficLight_Attribute_EAST;
+    if (obu_road_light1.road_direction() == 1) {
+      tl_attr = apollo::v2x::RoadTrafficLight_Attribute_EAST;
+    } else if (obu_road_light1.road_direction() == 2) {
+      tl_attr = apollo::v2x::RoadTrafficLight_Attribute_WEST;
+    } else if (obu_road_light1.road_direction() == 3) {
+      tl_attr = apollo::v2x::RoadTrafficLight_Attribute_SOUTH;
+    } else if (obu_road_light1.road_direction() == 4) {
+      tl_attr = apollo::v2x::RoadTrafficLight_Attribute_NORTH;
+    } else {
+      AINFO << "Road direction=" << obu_road_light1.road_direction()
+            << " is invalid.";
+    }
+    // FOR-EACH LANE
+    for (int idx_lane = 0; idx_lane < obu_road_light1.lane_trafficlight_size();
+         idx_lane++) {
+      const auto &obu_lane_light1 = obu_road_light1.lane_trafficlight(idx_lane);
+      if (!obu_lane_light1.has_gps_x_m() || !obu_lane_light1.has_gps_y_m()) {
+        AWARN << "Invalid lane_trafficlight: [" << idx_road << "][" << idx_lane
+              << "]: no gps info here.";
+        continue;
+      }
+      flag_has_data = true;
+      auto res_light1_ptr = res->add_road_traffic_light();
+      res_light1_ptr->set_gps_x_m(obu_lane_light1.gps_x_m());
```

Comment:
`_ptr` seems can be removed. You can use `T *result_light = result->add_road_traffic_light()` to indicate that it's a pointer.

And, Names should be descriptive; avoid abbreviation.
Always prefer long, human-readable variable/method/class names to short acronyms.

Ref: https://drake.mit.edu/styleguide/cppguide.html#General_Naming_Rules